### PR TITLE
Feature/86 create a lua scripting prototype with c

### PR DESCRIPTION
**Pull request by AmozPay**

### CODE REVIEW

**Comment by AmozPay** [comment]
just rebased changes on develop

File: OtterLib/scripting/include/ScriptingManager.hpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "lua_wrapper.hpp"
+#include "SystemManager.hpp"
+
+namespace Otter::Scripting {
+
+    static const std::string luaSystemsTable = "\
```
Comment (fzuefo) [inline OtterLib/scripting/include/ScriptingManager.hpp:8]: Comment for justifie the static const and explain the objectif

File: OtterLib/scripting/include/ScriptingManager.hpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            preUpdate = {},\
+            update = {},\
+            preDraw = {},\
+            draw = {},\
+            subDraw = {},\
+            [\"end\"] = {}\
+        }\
+    ";
+
+    static const std::string luaCallScripts = "\
```
Comment (fzuefo) [inline OtterLib/scripting/include/ScriptingManager.hpp:22]: need some doc

File: OtterLib/scripting/include/ScriptingManager.hpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        function __callScripts(phase)\n\
+            for k, v in pairs(OtterSystems[phase]) do\n\
+                v()\n\
+            end\n\
+        end\n\
+    ";
+
+    class ScriptingManager {
```
Comment (fzuefo) [inline OtterLib/scripting/include/ScriptingManager.hpp:30]: some docs

**Review by fzuefo — CHANGES_REQUESTED** [review]
some docs is need pls

**Review by fzuefo — APPROVED** [review]

**Pull request merged**